File: solutions/119.pascals-triangle-ii/pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        if(rowIndex == 0){
            List<Integer> result = new ArrayList<Integer>();
            result.add(1);
            return result;
        }
        if(rowIndex == 1){
            List<Integer> result = new ArrayList<Integer>();
            result.add(1);
            result.add(1);
            return result;
        }
        
        List<Integer> pre = new ArrayList<Integer>();
        pre.add(1);
        pre.add(1);
        
        for(int i = 2; i <= rowIndex; i++){ // i-th row
            List<Integer> temp = new ArrayList<Integer>();
            for(int j = 0; j < i; j++ ){ // j-th  item
                int value = pre.get(j) + (j == 0 ? 0 : pre.get(j-1));
                temp.add(value);
            }
            temp.add(1);
            pre = temp;
        }
        
        return pre;
    }
}

File: solutions/119.pascals-triangle-ii/pascals-triangle-ii.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    if(rowIndex === 0){
        return [1]
    }
    let pre = [1]
    for(let i = 0; i<= rowIndex; i++){
        let temp = []
        for(let j = 0; j < i; j++){
            temp[j] = pre[j] + (j === 0 ? 0 : pre[j-1])
        }
        temp[i] = 1
        pre = temp
    }
    return pre
};